paimon-core/src/main/java/org/apache/paimon/table/source/AbstractInnerTableScan.java

```diff
@@ -65,6 +69,16 @@ protected StartingScanner createStartingScanner(boolean isStreaming) {
             return new ContinuousCompactorStartingScanner();
         }
 
+        // read from consumer id
+        String consumerId = options.consumerId();
```

Comment:
do we need to compare the snapshot from consumer with snapshot from state?

Reply:
I think if the relevant content is recorded inside the state, it should not be read from the consumer file and should be trusted in the state.
Do you think something bad will happen?